src/MSBuild/ValidateMSBuildPackageDependencyVersions.cs

```diff
@@ -13,7 +13,7 @@ namespace MSBuild
     public class ValidateMSBuildPackageDependencyVersions : Task
     {
         [Required]
-        public string AppConfig { get; set; }
+        public ITaskItem AppConfig { get; set; }
```

Comment:
even though the class is public, it's not public api, right? 
since this would be a breaking change in that case 🤔

Reply:
The passed param is already item
https://github.com/dotnet/msbuild/blob/9a3cb6bbccea234ec7f2105ba6e1456d9f87b5ad/src/MSBuild/MSBuild.csproj#L312

The @(...) syntax means MSBuild was already passing an ITaskItem - it was just being implicitly converted to string before by engine.

Reply:
it looks like it happens here
https://github.com/dotnet/msbuild/blob/9a3cb6bbccea234ec7f2105ba6e1456d9f87b5ad/src/Build/BackEnd/TaskExecutionHost/TaskExecutionHost.cs#L779